src/Io/PsrAwaitRequestHandler.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace FrameworkX\Io;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use React\Http\Message\Response;
+use function React\Async\await;
+use function React\Promise\resolve;
+
+/**
+ * @internal
+ */
+class PsrAwaitRequestHandler implements RequestHandlerInterface
```

Comment:
This is mostly "stolen" from https://github.com/friends-of-reactphp/http-middleware-psr15-adapter and that needs attribution – I'm not sure how you usually handle this? As comment in the class or README?

Reply:
Normally we add a comment to the specific class, something like:

`# @copyright Copyright (c) 2024 Simon Frings, taken from https://github.com/clue/framework-x/pull/3 with permission`

Nearly all of these projects are published under the MIT license, same goes for https://github.com/friends-of-reactphp/http-middleware-psr15-adapter